src/main/scala/is/hail/methods/LinearRegression3.scala

```diff
@@ -77,10 +79,17 @@ object LinearRegression3 {
             i += 1
           }
 
+          // val dosageAC: DenseMatrix[Double] = sum(X(::, *))
+          val dosageAC: DenseVector[Double] = X.t(*, ::).map(r => sum(r))
+          assert(dosageAC.length == blockLength)
+
           val qtx: DenseMatrix[Double] = QtBc.value * X
```

Comment:
If you use this, add a test to the suite that will break if the first column is not constant.